Fix post sounding job

WalterKolczynski-NOAA opened the pull request:
**Description**
A few minor errors were blocking the sounding job from completing successfully:
- CFP APRUN command was using incorrect number of PEs
- Updates were needed to run on Orion. Removed the machine-specific code for Hera and Jet so it runs the same on all machines, while refactoring it to use C-style loop.
- Bash base errors (see also #1195), which are now corrected, while refactoring to use C-style loop.

This is part of a package of PRs that were tested together as part of a j-job refactor.

Refs #1195
Fixes #1211

**Type of change**
- [x] Bug fix (non-breaking change which fixes an issue)

**How Has This Been Tested?**
- [x] Cycled test on Orion
- [x] Cycled test on Hera
- [x] Cycled test on WCOSS2

**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

WalterKolczynski-NOAA committed:
Fix post sounding job

A few minor errors were blocking the sounding job from completing
successfully:
- CFP APRUN command was using incorrect number of PEs
- Updates were needed to run on Orion. Removed the machine-specific
code for Hera and Jet so it runs the same on all machines, while
refactoring it to use C-style loop.
- Bash base errors (see also #1195), which are now corrected, while
refactoring to use C-style loop.

Refs #1195
Fixes #1211

WalterKolczynski-NOAA commented:
Draft while I complete the rest of the PR group and get it tested on other machines. There may be a need to add the machine-specific section back in for WCOSS2 due to CFP differences.

KateFriedman-NOAA commented:
> There may be a need to add the machine-specific section back in for WCOSS2 due to CFP differences.

Perhaps, if needed, configure this at the env file level, similar to how the other CFP variables are set now (`mpmd_opt`, `USE_CFP` and `CFP_MP`)?

aerorahul reviewed (approved):
looks ok code-wise.  not sure I know how to test these.

WalterKolczynski-NOAA committed:
Add conditional for CFP CPU number in postsnd

The MPMD scripts on slurm machines need to be prefaced with the
processor rank(s) to be used, but PBS (WCOSS2) does not recognize
the ranks. Added the paradigm of other jobs of using `$CFP_MP` to
control whether rank numbers are added, then setting it to "YES"
on slurm machines.

Refs: #1211
